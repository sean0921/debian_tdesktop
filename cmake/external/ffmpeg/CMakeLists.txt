# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_ffmpeg INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_ffmpeg ALIAS external_ffmpeg)

if (DESKTOP_APP_USE_PACKAGED)
    find_package(PkgConfig REQUIRED)

    pkg_check_modules(AVCODEC REQUIRED libavcodec)
    pkg_check_modules(AVFORMAT REQUIRED libavformat)
    pkg_check_modules(AVUTIL REQUIRED libavutil)
    pkg_check_modules(SWSCALE REQUIRED libswscale)
    pkg_check_modules(SWRESAMPLE REQUIRED libswresample)

    target_include_directories(external_ffmpeg
    INTERFACE
        ${AVCODEC_INCLUDE_DIRS}
        ${AVFORMAT_INCLUDE_DIRS}
        ${AVUTIL_INCLUDE_DIRS}
        ${SWSCALE_INCLUDE_DIRS}
        ${SWRESAMPLE_INCLUDE_DIRS})

    target_link_libraries(external_ffmpeg
    INTERFACE
        ${AVCODEC_LIBRARIES}
        ${AVFORMAT_LIBRARIES}
        ${AVUTIL_LIBRARIES}
        ${SWSCALE_LIBRARIES}
        ${SWRESAMPLE_LIBRARIES})
else()
    target_include_directories(external_ffmpeg SYSTEM
    INTERFACE
        ${libs_loc}/ffmpeg
    )

    set(ffmpeg_lib_loc ${libs_loc}/ffmpeg)

    target_link_libraries(external_ffmpeg
    INTERFACE
        ${ffmpeg_lib_loc}/libavformat/libavformat.a
        ${ffmpeg_lib_loc}/libavcodec/libavcodec.a
        ${ffmpeg_lib_loc}/libswresample/libswresample.a
        ${ffmpeg_lib_loc}/libswscale/libswscale.a
        ${ffmpeg_lib_loc}/libavutil/libavutil.a
    )
    if (LINUX)
        target_link_static_libraries(external_ffmpeg
        INTERFACE
            va-x11
            va-drm
            va
            vdpau
            Xi
            Xext
            Xfixes
            Xrender
        )
        target_link_libraries(external_ffmpeg
        INTERFACE
          drm
        )
    endif()
endif()
