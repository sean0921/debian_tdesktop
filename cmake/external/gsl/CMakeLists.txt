# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_gsl INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_gsl ALIAS external_gsl)

target_include_directories(external_gsl SYSTEM
INTERFACE
    ${third_party_loc}/GSL/include
)
