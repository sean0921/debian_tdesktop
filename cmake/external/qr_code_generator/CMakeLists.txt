# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_qr_code_generator OBJECT)
add_library(desktop-app::external_qr_code_generator ALIAS external_qr_code_generator)
init_target(external_qr_code_generator "(external)")

set(qr_loc ${third_party_loc}/QR)
set(qr_src ${qr_loc}/cpp)

nice_target_sources(external_qr_code_generator ${qr_src}
PRIVATE
    BitBuffer.cpp
    BitBuffer.hpp
    QrCode.cpp
    QrCode.hpp
    QrSegment.cpp
    QrSegment.hpp
)

target_include_directories(external_qr_code_generator
PUBLIC
    ${qr_src}
)
