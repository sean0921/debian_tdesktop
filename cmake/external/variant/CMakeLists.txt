# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_variant INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_variant ALIAS external_variant)

target_include_directories(external_variant SYSTEM
INTERFACE
    ${third_party_loc}/variant/include
)
