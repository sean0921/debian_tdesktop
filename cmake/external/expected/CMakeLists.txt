# This file is part of Desktop App Toolkit,
# a set of libraries for developing nice desktop applications.
#
# For license and copyright information please follow this link:
# https://github.com/desktop-app/legal/blob/master/LEGAL

add_library(external_expected INTERFACE IMPORTED GLOBAL)
add_library(desktop-app::external_expected ALIAS external_expected)

target_include_directories(external_expected SYSTEM
INTERFACE
    ${third_party_loc}/expected/include
)
